corehq/plugins/signal_plugin.py

```diff
@@ -0,0 +1,41 @@
+from django.dispatch import Signal
+
+from corehq.plugins.interface import Plugins, ExtensionPoint
+from dimagi.utils.logging import log_signal_errors
+
+
+class PluginContribution:
+    def __init__(self, delegate):
+        self.delegate = delegate
+
+    def __call__(self, signal, sender, **kwargs):
+        # drop signal and sender
+        return self.delegate(**kwargs)
+
+
+class SignalPlugins(Plugins):
+    """Plugin implementation that uses Django signals to manage the
+    dispatching of extension point events
+    """
```

Comment:
Are you anticipating there would be other implementations of `Plugins`? What would be the use case for that?

Reply:
not necessarily but it does make it easier if we do